src/Transports.AspNetCore/GraphQLHttpMiddleware.cs

```diff
@@ -25,17 +32,32 @@ public class GraphQLHttpMiddleware<TSchema>
 
         private readonly RequestDelegate _next;
         private readonly PathString _path;
+
+#if NETSTANDARD2_0
         private readonly JsonSerializer _serializer;
 
         public GraphQLHttpMiddleware(RequestDelegate next, PathString path, Action<JsonSerializerSettings> configure)
+            : this(next, path)
         {
-            _next = next;
-            _path = path;
-
             var settings = new JsonSerializerSettings();
             configure(settings);
             _serializer = JsonSerializer.Create(settings); // it's thread safe https://stackoverflow.com/questions/36186276/is-the-json-net-jsonserializer-threadsafe
         }
+#else
+        private readonly JsonSerializerOptions _serializerOptions = new JsonSerializerOptions();
+
+        public GraphQLHttpMiddleware(RequestDelegate next, PathString path, Action<JsonSerializerOptions> configure)
+            : this(next, path)
+        {
+            configure(_serializerOptions);
+        }
+#endif
+
+        private GraphQLHttpMiddleware(RequestDelegate next, PathString path)
```

Comment:
I would prefer to inline this ctor.

Reply:
What do you mean by inline? I'm setting two fields so can't do `=> ...`

Reply:
I mean to remove third private ctor and assign fields inside two remaining ctors.

Reply:
Perhaps I should hold off on that. Should I introduce an interface for all this, something like 
`IGraphQLRequestDeserializer` and then introduce a project for NewtonsoftJson and SystemTextJson with an implementation?

Reply:
Gonna have the same pain as you guys did over in the core project where the `GraphQLRequest` can't be annotated, but I don't really wanna go through the pain of creating a ContractResolver or whatever it was. Maybe we can have an `IGraphQLRequest` and then each project can implement it with their own attributes. Otherwise I'm keen to just hardcode camelCase, or throw if that's not the configured setting and be done with it. What do you think?

Reply:
I'm still kind of failing to see the benefit of giving flexibility here just to deserialize a request body into a `GraphQLRequest`. If it was being used all over the shop, yea sure, but do we really have people restoring this as `netstandard2.0`? We'll surely drop that soon and just fall into netcore all the way at which point System.Text.Json comes free and we just use it, as is, to deserialize these requests.

Reply:
Anyway, will interface it and see where I end up :)

Reply:
1. ContractResolver <> pain, this is a very simple way to configure serializer.
2. Now `GraphQLRequest` tightly coupled with `Newtonsoft.Json` and must be refactored. 

> We'll surely drop that soon 

I'm not so sure.

As an option, we can put the `ISomeSerializer<T>` interface in the `GraphQL-dotnet` project itself, and implementations in the corresponding `GraphQL.SystemTextJson` and `GraphQL.NewtonsoftJson` project so as not to spawn new small projects. And all serialization will be concentrated in the project intended for this - be it some entity from core graphql-dotnet or from server project (as `GraphQLRequest` does).

Reply:
This should not worsen the degree of decomposition of responsibility between projects.

Reply:
I've got somewhat of a plan fwd, matches what you're thinking, so I'll get to work and we'll review a bit later. Can only spend a bit more time on it tonight though. Thanks for the help :)

Reply:
You can just wait for the #1448 in graphql repo. After that it may become clearer where and how to move.

Reply:
Going to resolve this so we can start a new conversation around what I've stubbed out now behind interface/s.